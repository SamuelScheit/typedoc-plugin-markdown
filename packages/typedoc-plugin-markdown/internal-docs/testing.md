---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
