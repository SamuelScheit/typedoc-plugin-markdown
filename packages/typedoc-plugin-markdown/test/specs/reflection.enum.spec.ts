import { expectFileToEqual } from '@devtools/testing';

describe(`Enum Reflection`, () => {
  test(`should compile basic enum`, () => {
    expectFileToEqual('reflections', 'members', 'enumerations/BasicEnum.md');
  });

  test(`should compile enum with values`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      '/enumerations/EnumWithValues.md',
    );
  });
});
