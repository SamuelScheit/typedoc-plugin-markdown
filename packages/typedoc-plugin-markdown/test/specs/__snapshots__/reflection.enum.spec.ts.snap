// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enum Reflection should compile basic enum: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Enumeration: BasicEnum

Comments for enum

## Enumeration Members

### MemberA

> **MemberA**: \`0\`

Comments for MemberA

#### Source

[enums.ts:12](http://source-url)

***

### MemberB

> **MemberB**: \`1\`

Comments for MemberB

#### Source

[enums.ts:16](http://source-url)
"
`;

exports[`Enum Reflection should compile basic enum: (Output File Strategy "members") (Option Group "2") 1`] = `
"# BasicEnum

Comments for enum

## Enumeration Members

| Enumeration Member | Value | Description |
| :------ | :------ | :------ |
| \`MemberA\` | \`0\` | Comments for MemberA |
| \`MemberB\` | \`1\` | Comments for MemberB |
"
`;

exports[`Enum Reflection should compile enum with values: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Enumeration: EnumWithValues

Comments for enum

## Enumeration Members

### MemberA

> **MemberA**: \`"UP"\`

#### Source

[enums.ts:23](http://source-url)

***

### MemberB

> **MemberB**: \`"DOWN"\`

#### Source

[enums.ts:24](http://source-url)
"
`;

exports[`Enum Reflection should compile enum with values: (Output File Strategy "members") (Option Group "2") 1`] = `
"# EnumWithValues

Comments for enum

## Enumeration Members

| Enumeration Member | Value |
| :------ | :------ |
| \`MemberA\` | \`"UP"\` |
| \`MemberB\` | \`"DOWN"\` |
"
`;
